File: src/main/java/ProbYDanioVerdadero.java
public class ProbYDanioVerdadero extends DecoratorAtaques{
    private final double probabilidad;
    private final double porcentajeDanioVerdadero;

    ProbYDanioVerdadero(AtaqueConcret aDecorar,double probabilidad,double porcentajeDanioVerdadero) {
        super(aDecorar);
        this.probabilidad = probabilidad * 0.01;
        this.porcentajeDanioVerdadero = porcentajeDanioVerdadero * 0.01;
    }
    int especialEfecto(Chinpokomon atacado){
        int resultado;
            if( Math.random() <= this.probabilidad){
            resultado = (int) (atacado.vidaActual * this.porcentajeDanioVerdadero);
        }else{
            resultado = this.getDanio();
        }
        return resultado;
    }

    public boolean tieneEfecto() { return true;}
}

File: python/ProbYDanioVerdadero.py
from math import floor
import random
from DecoratorAtaques import DecoratorAtaques
from Chimpokomon import Chinpokomon

class ProbYDanioVerdadero(DecoratorAtaques):

    def __init__(self, aDecorar, probabilidad, porcentajeDanioVerdadero) :
        super().__init__(aDecorar)
        self.probabilidad = probabilidad * 0.01
        self.porcentajeDanioVerdadero = porcentajeDanioVerdadero * 0.01
    
    def especial_efecto(self, atacado):
        resultado = 0
        if random.random() <= self.probabilidad:
            resultado = floor(atacado.get_vida_actual() * self.porcentajeDanioVerdadero)
        else:
            resultado = self.get_danio()
        
        return resultado
    

    def tiene_efecto(self):
        return True